### PR TITLE
Fix Github CI tests not ran

### PR DESCRIPTION
**Pull request by weilueluo**
https://jira-eng-rtp3.cisco.com/jira/browse/LH-69739 (or Github issue ID)

### Description
- Make github CI actually run the tests.
- Fix a bunch of inconsistent naming, we should be using `ConnectorName`, `ConnectorType`, and `SocketAddress`, instead of `sdcName`, `sdcType`, and `ipv4`.
- Acceptance test matrix for different version of terraform now runs sequentially. (this means much longer to run >:)
- Fix the hinting generate docs command when docs is missing.
- Incorporate more lenient retries (by ignore error until max retries reached) for all requests made through internal http package because our backend api is flaky and acceptance tests keep failing if it is too strict.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -113,5 +115,5 @@ jobs:
       - env:
           TF_ACC: "1"
           ACC_TEST_CISCO_CDO_API_TOKEN: ${{ secrets.ACC_TEST_CISCO_CDO_API_TOKEN }}
-        run: go test -v -cover ./internal/provider/
+        run: go test -v -cover ./...
```
Comment (weilueluo) [inline .github/workflows/test.yml:118]: make it covers all the test files under provider directory

File: client/internal/http/request.go [hunk 2]
```diff
@@ -94,7 +95,7 @@ func (r *Request) send(output any) error {
 	// check status
 	if res.StatusCode >= 400 {
 		body, err := io.ReadAll(res.Body)
-		err = fmt.Errorf("failed: code=%d, status=%s, body=%s, readBodyErr=%s", res.StatusCode, res.Status, string(body), err)
+		err = fmt.Errorf("failed: code=%d, status=%s, body=%s, readBodyErr=%s, url=%s, method=%s", res.StatusCode, res.Status, string(body), err, r.url, r.method)
```
Comment (weilueluo) [inline client/internal/http/request.go:98]: added a bit more debugging information, url and method should be fine as it does not contain sensitive information

File: client/internal/retry/do.go [hunk 3]
```diff
@@ -21,6 +22,8 @@ type Options struct {
 	Retries int           // Retries is the max number of retries before terminating. Negative means no limit.
 
 	Logger *log.Logger
+
+	IgnoreError bool // IgnoreError will ignore and save error returned from Func, when max retries exceeded, errors are combined the returned together.
```
Comment (weilueluo) [inline client/internal/retry/do.go:26]: Added this to improve robustness, because our aegis api is flaky and not eventual consistent

Comment (bobertyy) [reply client/internal/retry/do.go:26]: I think we should rename this - we don't actually ignore the errors but instead prevent an early termination from them.

What about `PreventEarlyExitOnError`?

Comment (weilueluo) [reply client/internal/retry/do.go:26]: Yea, make sense, did not think too much when writing it down, let's use `EarlyExitOnError`? `Prevent` seems slightly redundant

**Review by hazital — APPROVED** [review]
LGTM nice, we just need to ensure that the acceptance tests are passed before merging

**Pull request merged**